### PR TITLE
Updated license text for Apollo, added SPDX to headers/source.

### PR DESCRIPTION
**Pull request by cdwdirect**
Primary purpose:
- Copyright and license notices updated.
- Followed the example of the Spack repository, per @tgamblin 's suggestion in previous PR.

Additionally:
- Removed outdated / unnecessary files to tidy up the top-level folder.
- Moved SOS+Python tools and examples into the `archive` folder for reference.
- Trailing spaces trimmed out automatically by vim during editing, across numerous files.

### CODE REVIEW

File: COPYRIGHT [hunk 1]
```diff
@@ -0,0 +1,23 @@
+Intellectual Property Notice
+------------------------------
+
+Apollo is licensed under the Apache License, Version 2.0 (LICENSE-APACHE
+or http://www.apache.org/licenses/LICENSE-2.0) or the MIT license,
+(LICENSE-MIT or http://opensource.org/licenses/MIT), at your option.
+
+Copyrights and patents in the Apollo project are retained by contributors.
+No copyright assignment is required to contribute to Apollo.
```
Comment (tgamblin) [inline COPYRIGHT]: We cannot just switch the license of Apollo from `MIT` to `Apache-2.0 OR MIT`.  It has to stay MIT, unless we want to send it through the process again AND get all contributors' permission, so please change all these `Apache-2.0 OR MIT`'s to `MIT`.

For a single-licensed project, we do't need all this fanciness so also remove `COPYRIGHT`.

File: LICENSE-APACHE [hunk 2]
```diff
@@ -0,0 +1,202 @@
+
+                                 Apache License
```
Comment (tgamblin) [inline LICENSE-APACHE]: Remove this file and use a single MIT `LICENSE` file.

File: LICENSE-MIT [hunk 3]
```diff
@@ -0,0 +1,21 @@
+MIT License
```
Comment (tgamblin) [inline LICENSE-MIT]: This should just go in a file called `LICENSE`.

File: README.md [hunk 4]
```diff
@@ -27,12 +27,18 @@ the following paper:
 
 ## License
 
-Apollo is distributed under the terms of both the MIT license.  All new
-contributions must be made under the MIT license.
+Apollo is distributed under the terms of both the MIT license and the
+Apache License (Version 2.0). Users may choose either license, at their
+option.
```
Comment (tgamblin) [inline README.md]: no - MIT

File: README.md [hunk 5]
```diff
@@ -27,12 +27,18 @@ the following paper:
...
 
-See [LICENSE](https://github.com/LLNL/apollo/blob/master/LICENSE) and
-[NOTICE](https://github.com/LLNL/apollo/blob/master/NOTICE) for details.
+All new contributions must be made under both the MIT and Apache-2.0
+licenses.
```
Comment (tgamblin) [inline README.md]: Add this sentence here:

> Copyrights and patents in the Apollo project are retained by contributors. No copyright assignment is required to contribute to Apollo.

File: README.md [hunk 6]
```diff
@@ -27,12 +27,18 @@ the following paper:
...
 
-SPDX-License-Identifier: MIT
+See [LICENSE-MIT](https://github.com/LLNL/apollo/blob/develop/LICENSE-MIT),
+[LICENSE-APACHE](https://github.com/LLNL/apollo/blob/develop/LICENSE-APACHE),
+[COPYRIGHT](https://github.com/LLNL/apollo/blob/develop/COPYRIGHT), and
+[NOTICE](https://github.com/LLNL/apollo/blob/develop/NOTICE) for details.
+
+SPDX-License-Identifier: (Apache-2.0 OR MIT)
```
Comment (tgamblin) [inline README.md]: Just MIT.

**Review by tgamblin — CHANGES_REQUESTED** [review]
@ggeorgakoudis should review the technical side of this, but on the license stuff: headers are good, but we can't change the license of Apollo.  It's MIT licensed.

**Comment by cdwdirect** [comment]
> @ggeorgakoudis should review the technical side of this, but on the license stuff: headers are good, but we can't change the license of Apollo. It's MIT licensed.

Thanks for the head's up! I've corrected the license issues you indicated, pushed to `c4b99af` and `499f1b4` commits above.

**Comment by ggeorgakoudis** [comment]
Superseded by recent development

**Pull request closed without merging**